File: app/src/main/java/com/billpayments/features/payments/ui/ReceiptScreenData.kt
package com.billpayments.features.payments.ui

import androidx.compose.runtime.Immutable
import java.time.Instant

@Immutable
sealed interface ReceiptScreenData {
    data object NotFound : ReceiptScreenData
    data class Processing(val billerName: String, val amountPaise: Long) : ReceiptScreenData
    data class Success(
        val paymentId: String,
        val billerId: String,
        val billerName: String,
        val account: String,
        val amountPaise: Long,
        val paidAt: Instant,
        val canSaveBiller: Boolean,
    ) : ReceiptScreenData
    data class Failed(
        val billerId: String,
        val billerName: String,
        val categoryId: String,
        val account: String,
        val amountPaise: Long,
    ) : ReceiptScreenData
}

File: BillPayments/Features/Payments/UI/ReceiptScreenData.swift
import Foundation

enum ReceiptScreenData: Equatable {
    case notFound
    case processing(billerName: String, amountPaise: Int64)
    case success(
        paymentId: String,
        billerId: String,
        billerName: String,
        account: String,
        amountPaise: Int64,
        paidAt: Date,
        canSaveBiller: Bool
    )
    case failed(
        billerId: String,
        billerName: String,
        categoryId: String,
        account: String,
        amountPaise: Int64
    )
}
